modules/other-integrations/pages/execute-dbt-teradata-transformations-in-airflow-with-cosmos.adoc

```diff
@@ -0,0 +1,247 @@
++
+NOTE: Change `/../../` to path of jaffle_shop project path.
+
+3. Define the virtual environment path where dbt-teradata installed in <<Install dbt>> as an environment variable `dbt_venv_dir`.
+[source, bash]
+set dbt_venv_dir=/../../dbt_env/bin/dbt
++
+NOTE: Change `/../../` to path where virtual environment path defined.
+
+== Start Apache Airflow web server
+1. Run airflow web server
++
+[source, bash]
+----
+airflow standalone
+----
+2. Access the airflow UI. Visit https://localhost:8080 in the browser and log in with the admin account details shown in the terminal.
+
+== Define Apache Airflow connection to Vantage Cloud Lake
+
+1. Click on Admin - Connections
+2. Click on + to define new connection to Teradata vantage cloud lake instance.
+3. Define new connection with id `teradata_default` with Teradata vantage cloud lake instance details.
+* Connection Id: teradata_lake
+* Connection Type: Teradata.
+* Database Server URL (required): Teradata vantage cloud lake instance hostname to connect to.
+* Database: jaffle_shop
+* Login (required): lake_user
+* Password (required): lake_user
+
+== Define DAG in Apache Airflow
+Dags in airflow will be defined as python file. Similarly, define below dag to run dbt transformations defined in jaffle_shop dbt project against Teradata Vantage system using cosmos. Copy below python code and save it as `airflow-cosmos-dbt-teradata-integration.py` under directory $AIRFLOW_HOME/files/dags.
+[source, python]
+----
+import os
+from datetime import datetime
+
+from airflow import DAG
+from cosmos import DbtTaskGroup, ProjectConfig, ProfileConfig, ExecutionConfig
+from cosmos.profiles import TeradataUserPasswordProfileMapping
+
+PATH_TO_DBT_VENV = f"{os.environ['dbt_venv_dir']}"
+PATH_TO_DBT_PROJECT = f"{os.environ['dbt_project_home_dir']}"
+
+
+execution_config = ExecutionConfig(
+    dbt_executable_path=PATH_TO_DBT_VENV,
+)
+profile_config = ProfileConfig(
+    profile_name="generated_profile",
+    target_name="dev",
+    profile_mapping=TeradataUserPasswordProfileMapping(
+        conn_id="teradata_default",
+    ),
+)
+with DAG(
+    dag_id="execute_dbt_transformations_with_cosmos",
+    max_active_runs=1,
+    max_active_tasks=10,
+    catchup=False,
+    start_date=datetime(2024, 1, 1),
+
+) as dag:
+      transform_data = DbtTaskGroup(
+          group_id="transform_data",
+          project_config=ProjectConfig(PATH_TO_DBT_PROJECT),
+          profile_config=profile_config,
+          execution_config=execution_config,
+          default_args={"retries": 2},
+      )
+----
+
+== Load DAG
+
+When the dag file is copied to $AIRFLOW_HOME/files/dags, apache airflow loads the dag to airflow UI.
+
+== Run DAG
+
+Run the dag as shown in below image.
+
+image::{dir}/airflow-dag.png[Run dag,align="left" width=75%]
+
+== Summary
+
+In this quick start guide, we explored how to utilize Astronomer Cosmos library in apache airflow to execute dbt teradata transformations against Teradata Vantage instance.
```

Comment:
dbt teradata transformations should be replaced with _dbt transformations_

Reply:
agree with this

Reply:
Modified